core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -355,6 +358,38 @@ public void removeBlock(long sessionId, long blockId)
     mBlockStore.removeBlock(sessionId, blockId);
   }
 
+  public void freeWorker() throws IOException {
+    List<String> paths = new ArrayList<>();
+    if (Configuration.global().get(PropertyKey.WORKER_BLOCK_STORE_TYPE) == BlockStoreType.FILE) {
+      int tierCount = Configuration.global().getInt(PropertyKey.WORKER_TIERED_STORE_LEVELS);
+      for (int i = 0; i < tierCount; i++) {
+        paths.addAll(Configuration.global().getList(PropertyKey
+                .Template.WORKER_TIERED_STORE_LEVEL_DIRS_PATH.format(i)));
+      }
+    }
+    else
+      paths.addAll(Configuration.global().getList(PropertyKey.WORKER_PAGE_STORE_DIRS));
+
+    List<String> failDeleteDirs = new ArrayList<>();
+    for (String tmpPath : paths) {
+        File[] files = new File(tmpPath).listFiles();
+        Preconditions.checkNotNull(files, "The path does not denote a directory.");
+        List<String> subDirectories = new ArrayList<>();
+        for (File file : files)
+          subDirectories.add(file.getPath());
+        for (String s : subDirectories) {
+          try {
+            FileUtils.deletePathRecursively(s);
+          } catch (IOException ie) {
+            failDeleteDirs.add(s);
+          }
+        }
+    }
+    if (!failDeleteDirs.isEmpty())
+      throw new IOException("These directories are failed to be freed: " + failDeleteDirs);
```

Comment:
throw AlluxioRuntimeException, add throwable param.

Reply:
> throw AlluxioRuntimeException, add throwable param.

AlluxioRuntimeException will be also transmitted to other exception in `RpcUtils.callAndReturn` method.